### PR TITLE
Focus Mode 1.1 "mouseleave" logic fix

### PR DESCRIPTION
**Pull request by bvaughn**
Fixes https://github.com/RecordReplay/customer-support/issues/89

### CODE REVIEW

File: src/ui/components/Timeline/Focuser.tsx [hunk 1]
```diff
@@ -125,7 +125,17 @@ function Focuser({ editMode, setEditMode }: Props) {
     };
 
     // Stop all drag operations when the mouse leaves the window.
-    const onWindowMouseLeave = () => {
+    const onDocumentMouseLeave = (event: MouseEvent) => {
```
Comment (bvaughn) [inline src/ui/components/Timeline/Focuser.tsx:128]: Looking at the Replay, "mouseleave" was being dispatched for things other than the outer window/document. This is not how I expected it to work (or how it worked for me on MacOS + Chrome during testing) but it seems like _at least one fix_ here is to bounds check before assuming the mouse has _actually_ left the document.

Testing on Chrome, Firefox, and Safari seems good for me after this change.

@markerikson Can you verify on Windows?

Comment (jcmorrow) [reply src/ui/components/Timeline/Focuser.tsx:128]: Woah, weird but good thinking!

**Review by jcmorrow — COMMENTED** [review]
I'll wait for Mark to 👍 on the windows fix but LGM

File: src/ui/components/Timeline/Focuser.tsx (hunk 1 above)
Comment (markerikson) [reply src/ui/components/Timeline/Focuser.tsx:128]: Seems smooth to me on both Chrome and FF!

**Review by markerikson — APPROVED** [review]

**Pull request merged**